code/controllers/subsystem/weather.dm

```diff
@@ -82,7 +82,7 @@ SUBSYSTEM_DEF(weather)
     var/datum/weather/A
     for(var/V in processing)
         var/datum/weather/W = V
-        if((z in W.impacted_z_levels) && W.area_type == active_area.type)
+        if((z in W.impacted_z_levels) && W.area_type == active_area)
```

Comment:
So this has been broken all this time?

Reply:
I guess, the proc has never been used in the project before though. I was trying to figure out how to get it a darn instance of an area in Lava land but then I realized that it doesn't really make sense to give it an instance because the actual weather itself has a target area _type_